aclk/agent_cloud_link.c

```diff
@@ -1214,7 +1214,7 @@ inline void aclk_create_header(BUFFER *dest, char *type, char *msg_id, time_t ts
 
     buffer_sprintf(
         dest,
-        "\t{\"type\": \"%s\",\n"
+        "{\t\"type\": \"%s\",\n"
```

Comment:
I wouldn't move the tab after the curly bracket though (i.e. just `{"type":`). If possible at all it would be great to just remove all unnecessary whitespaces, if that is not possible we can leave it I guess. It's not a big deal but the reason I'm asking is that I'm running a regular expression to detect whether a message is an `hello` one or not. I can account for all possible whitespaces in the regex of course but I thought it'd be better to ask for a change to save some bandwidth in the communication as well.

Reply:
I am all for a minified JSON in ACLK. Actually this has been my stance since the beginning. There are plenty of tools to pretty-print JSON for humanz where needed.
Even during development, I use test server/tools in Ruby and they do pretty print for me where needed.

Reply:
@fracasula This fixes the problem so that you can skip the whitespace processing step on your side.

If @stelfrag and @mfundul don't have an issue with sending minified JSON I will then think of minifying everything at once with one PR.
There is a bit of caveat on our side because we use `/api/v1/info` and embed it into our JSON and that is not minified as well.

Reply:
Let's wait for the minify then and close this PR? I don't mind having an extra check now if we can have a minified version later. Consider that all messages get debounced as well so they end up in Redis back and forth and then in Pulsar where we have considerable retention enabled as well. It would save us quite some space in the long run.

Reply:
well, they are not mutually exclusive.
this removes the leading whitespace problem, next PR will minify

Reply:
True but the minification should remove the trailing whitespace too so I proposed we postpone since it's a minor issue like you presented it in the title of this PR. I'm fine either way.

Reply:
Don't see any advantage in waiting. It will just delay when you can remove the needless code on the cloud side, doesn't really give us any advantage